### PR TITLE
Removed custom split method as it is not effective anymore

### PR DESCRIPTION
**Pull request by doom369**
Motivation

Starting from Java 7 method `String.split` in java [is optimized for splitting by single char string](http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/7-b147/java/lang/String.java#String.split%28java.lang.String%2Cint%29). Thus custom `StringUtil.split` method doesn't have any sense anymore. Even more - it slower than optimized java version.

Benchmark results:
```
Benchmark                                                                                          (s)   Mode  Cnt      Score       Error   Units
StringUtilPerfTest.customSplit                                                                      aw  thrpt   10  54943.557 ±  6337.244  ops/ms
StringUtilPerfTest.customSplit                                                              someString  thrpt   10  44412.726 ±  5632.855  ops/ms
StringUtilPerfTest.customSplit                                                someString:someParameter  thrpt   10   9412.291 ±  1062.826  ops/ms
StringUtilPerfTest.customSplit                                  someString:someParameter:someParameter  thrpt   10   6626.128 ±   877.140  ops/ms
StringUtilPerfTest.customSplit  someStringsomeStringsomeString:someParametersomeParametersomeParameter  thrpt   10   5655.373 ±   914.399  ops/ms
StringUtilPerfTest.split                                                                            aw  thrpt   10  72180.120 ± 10025.371  ops/ms
StringUtilPerfTest.split                                                                    someString  thrpt   10  61568.319 ±  9280.446  ops/ms
StringUtilPerfTest.split                                                      someString:someParameter  thrpt   10   9796.448 ±  1103.540  ops/ms
StringUtilPerfTest.split                                        someString:someParameter:someParameter  thrpt   10   7475.525 ±   863.184  ops/ms
StringUtilPerfTest.split        someStringsomeStringsomeString:someParametersomeParametersomeParameter  thrpt   10   8538.443 ±   795.152  ops/ms
```

Benchmark : 

```
@BenchmarkMode(Mode.Throughput)
@OutputTimeUnit(TimeUnit.MILLISECONDS)
@State(Scope.Thread)
@Fork(1)
@Warmup(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
@Measurement(iterations = 10, time = 1, timeUnit = TimeUnit.SECONDS)
public class StringUtilPerfTest {

    @Param({"aw", "someString", "someString:someParameter", "someString:someParameter:someParameter", "someStringsomeStringsomeString:someParametersomeParametersomeParameter"})
    public String s;

    @Benchmark
    public String[] split() {
        return s.split(":");
    }

    @Benchmark
    public String[] customSplit() {
        return StringUtil.split(s, ':');
    }
}

```

Modifications:
Replaced all occurrences of `StringUtil.split` with `String.split`.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@doom369 we can only use the provided split method if using java7

**Comment by doom369** [comment]
@normanmaurer it works for any java version. But for Java version < 7 `String.split` is not optimized for single char splitting as I know However split changes could be back ported. Need to check.

**Comment by normanmaurer** [comment]
@doom369 I see.. thanks

File: codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsRecordEncoder.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 import io.netty.util.internal.StringUtil;
 import io.netty.util.internal.UnstableApi;
 
-import static io.netty.handler.codec.dns.DefaultDnsRecordDecoder.ROOT;
+import static io.netty.handler.codec.dns.DefaultDnsRecordDecoder.*;
```
Comment (normanmaurer) [inline codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsRecordEncoder.java]: @doom369 revert please

File: codec-http/src/main/java/io/netty/handler/codec/http/CookieDecoder.java [hunk 2]
```diff
@@ -15,20 +15,14 @@
...
 
 import java.text.ParseException;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
-import java.util.Set;
-import java.util.TreeSet;
+import java.util.*;
+
+import static io.netty.handler.codec.http.CookieUtil.*;
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/CookieDecoder.java]: @doom369 revert these import changes please.

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaders.java [hunk 3]
```diff
@@ -19,18 +19,12 @@
...
-import java.util.Date;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 import java.util.Map.Entry;
-import java.util.Set;
 
-import static io.netty.util.internal.ObjectUtil.checkNotNull;
+import static io.netty.util.internal.ObjectUtil.*;
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaders.java]: @doom369 revert import changes please.

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java [hunk 4]
```diff
@@ -16,12 +16,7 @@
 package io.netty.handler.codec.http.multipart;
 
 import io.netty.buffer.ByteBuf;
-import io.netty.handler.codec.http.HttpConstants;
-import io.netty.handler.codec.http.HttpContent;
-import io.netty.handler.codec.http.HttpHeaderNames;
-import io.netty.handler.codec.http.HttpHeaderValues;
-import io.netty.handler.codec.http.HttpRequest;
-import io.netty.handler.codec.http.LastHttpContent;
+import io.netty.handler.codec.http.*;
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java]: @doom369 revert import changes please

File: codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketClientHandshaker.java [hunk 5]
```diff
@@ -15,26 +15,9 @@
...
-import io.netty.handler.codec.http.HttpContentDecompressor;
-import io.netty.handler.codec.http.HttpHeaderNames;
-import io.netty.handler.codec.http.HttpHeaders;
-import io.netty.handler.codec.http.HttpObjectAggregator;
-import io.netty.handler.codec.http.HttpRequestEncoder;
-import io.netty.handler.codec.http.HttpResponse;
-import io.netty.handler.codec.http.HttpResponseDecoder;
-import io.netty.handler.codec.http.HttpScheme;
+import io.netty.channel.*;
+import io.netty.handler.codec.http.*;
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketClientHandshaker.java]: revert import changes

File: codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketServerHandshaker.java [hunk 6]
```diff
@@ -15,25 +15,10 @@
...
-import io.netty.handler.codec.http.FullHttpResponse;
-import io.netty.handler.codec.http.HttpContentCompressor;
-import io.netty.handler.codec.http.HttpHeaders;
-import io.netty.handler.codec.http.HttpObjectAggregator;
-import io.netty.handler.codec.http.HttpRequest;
-import io.netty.handler.codec.http.HttpRequestDecoder;
-import io.netty.handler.codec.http.HttpResponseEncoder;
-import io.netty.handler.codec.http.HttpServerCodec;
+import io.netty.channel.*;
+import io.netty.handler.codec.http.*;
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketServerHandshaker.java]: revert import changes

File: codec-http/src/main/java/io/netty/handler/codec/http/websocketx/extensions/WebSocketExtensionUtil.java [hunk 7]
```diff
@@ -18,21 +18,16 @@
...
 import io.netty.handler.codec.http.HttpHeaderValues;
 import io.netty.handler.codec.http.HttpHeaders;
-import io.netty.util.internal.StringUtil;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/websocketx/extensions/WebSocketExtensionUtil.java]: revert import changes

File: codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java [hunk 8]
```diff
@@ -24,13 +24,11 @@
...
 import io.netty.handler.codec.stomp.StompSubframeDecoder.State;
 import io.netty.util.internal.AppendableCharSequence;
-import io.netty.util.internal.StringUtil;
 
 import java.util.List;
 import java.util.Locale;
 
-import static io.netty.buffer.ByteBufUtil.indexOf;
-import static io.netty.buffer.ByteBufUtil.readBytes;
+import static io.netty.buffer.ByteBufUtil.*;
```
Comment (normanmaurer) [inline codec-stomp/src/main/java/io/netty/handler/codec/stomp/StompSubframeDecoder.java]: revert import changes

File: handler-proxy/src/test/java/io/netty/handler/proxy/HttpProxyServer.java [hunk 9]
```diff
@@ -23,19 +23,8 @@
...
-import io.netty.handler.codec.http.HttpMethod;
-import io.netty.handler.codec.http.HttpObjectAggregator;
-import io.netty.handler.codec.http.HttpRequestDecoder;
-import io.netty.handler.codec.http.HttpResponseEncoder;
-import io.netty.handler.codec.http.HttpResponseStatus;
-import io.netty.handler.codec.http.HttpServerCodec;
-import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.codec.http.*;
 import io.netty.util.CharsetUtil;
-import io.netty.util.internal.StringUtil;
```
Comment (normanmaurer) [inline handler-proxy/src/test/java/io/netty/handler/proxy/HttpProxyServer.java]: revert import changes

File: transport-native-epoll/src/test/java/io/netty/channel/epoll/EpollReuseAddrTest.java [hunk 10]
```diff
@@ -18,16 +18,11 @@
 import io.netty.bootstrap.AbstractBootstrap;
 import io.netty.bootstrap.Bootstrap;
 import io.netty.bootstrap.ServerBootstrap;
-import io.netty.channel.ChannelFuture;
-import io.netty.channel.ChannelHandler;
-import io.netty.channel.ChannelHandlerAdapter;
-import io.netty.channel.ChannelHandlerContext;
-import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.*;
```
Comment (normanmaurer) [inline transport-native-epoll/src/test/java/io/netty/channel/epoll/EpollReuseAddrTest.java]: revert import changes

**Comment by doom369** [comment]
@normanmaurer reverted. sorry for mess.

**Comment by trustin** [comment]
Nice! :-)

**Comment by normanmaurer** [comment]
@doom369 can you please squash your commits

**Comment by doom369** [comment]
@normanmaurer that was hard, but I did it :)

**Comment by normanmaurer** [comment]
Cherry-picked into 4.0 (68ee85f1ae0c985cb59618d33416471c0320877a) and 4.1 (a80ea46b8e6d3b225547ae138a357101b0381268)

**Pull request closed without merging**